src/main/java/at/hannibal2/skyhanni/data/IslandGraphs.kt

```diff
@@ -260,64 +296,185 @@ object IslandGraphs {
+        // when a faster node with the same name + tags exists, reroute to this node
+        allowRerouting: Boolean = false,
+        condition: () -> Boolean = { true },
+    ) {
+        reset()
+        currentTargetNode = this
+        shouldAllowRerouting = allowRerouting
+        pathFind0(location = position, label, color, onFound, showGoalExact = false, condition)
     }
 
     fun pathFind(
         location: LorenzVec,
+        label: String,
         color: Color = LorenzColor.WHITE.toColor(),
         onFound: () -> Unit = {},
+        // when the goal is further away from the node, show a line and waypoint to that node
         showGoalExact: Boolean = false,
         condition: () -> Boolean = { true },
     ) {
         reset()
+        shouldAllowRerouting = false
+        pathFind0(location, label, color, onFound, showGoalExact, condition)
+    }
+
+    private fun pathFind0(
+        location: LorenzVec,
+        label: String,
+        color: Color = LorenzColor.WHITE.toColor(),
+        onFound: () -> Unit = {},
+        showGoalExact: Boolean = false,
+        condition: () -> Boolean = { true },
+    ) {
         currentTarget = location
+        this.label = label
         this.color = color
         this.onFound = onFound
         this.showGoalExact = showGoalExact
         this.condition = condition
         val graph = currentIslandGraph ?: return
         goal = graph.minBy { it.position.distance(currentTarget!!) }
+        updateChat()
+    }
+
+    private const val PATHFIND_ID = -6457563
+
+    private fun updateChat() {
+        if (label == "") return
+        val finalDistance = distanceViaNodes + distanceToNextNode
+        if (finalDistance == 0.0) return
+        val distance = finalDistance.roundTo(1)
+        if (totalDistance == 0.0 || distance > totalDistance) {
+            totalDistance = distance
+        }
+        senChatDistance(distance)
+    }
+
+    private fun senChatDistance(distance: Double) {
+        val percentage = (1 - (distance / totalDistance)) * 100
+        val componentText = "§e[SkyHanni] Navigating to §r$label §f[§e$distance§f] §f(§c${percentage.roundTo(1)}%§f)".asComponent()
+        componentText.onClick(
+            onClick = {
+                stop()
+                "§e[SkyHanni] Navigation manually stopped!".asComponent().send(PATHFIND_ID)
+            },
+        )
+        componentText.hover = "§eClick to stop navigating!".asComponent()
+        componentText.send(PATHFIND_ID)
     }
 
     @SubscribeEvent
     fun onRenderWorld(event: LorenzRenderWorldEvent) {
         if (!LorenzUtils.inSkyBlock) return
-        val path = fastestPath ?: return
+        var path = fastestPath ?: return
 
-        var graph = path
-        graph = skipNodes(graph) ?: graph
+        if (path.nodes.size > 1) {
+            val hideNearby = if (Minecraft.getMinecraft().thePlayer.onGround) 5 else 7
+            path = Graph(path.nodes.filter { it.position.distanceToPlayer() > hideNearby })
```

Comment:
takeLastWhile would be more performant with the same result. (The predicate just needs to be inverted)

Reply:
replacing this with `path = Graph(path.nodes.takeLastWhile { it.position.distanceToPlayer() < hideNearby })` doesnt work.

Reply:
A the predicate does not need to be inverted. ups.